### PR TITLE
IRIS-4911 | Refactor SortingTests, possibly add more test cases and get rid of deprecated component Loading

### PR DESCRIPTION
**Pull request by tortila**
[Ticket](https://wikia-inc.atlassian.net/browse/IRIS-4911)

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java [hunk 1]
```diff
@@ -4,10 +4,42 @@
...
+  private final DiscussionsHeader discussionsHeader = new DiscussionsHeader();
+
+  @Getter(lazy = true)
+  private final FiltersPopOver filtersPopOver = new FiltersPopOver();
+
+  @Getter(lazy = true)
+  private final Promoting promoting = new Promoting();
+
+  @Getter(lazy = true)
+  private final CommunityBadge communityBadge = new CommunityBadge();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java [hunk 2]
```diff
@@ -4,10 +4,42 @@
...
+
+  @Getter(lazy = true)
+  private final HeroUnit heroUnit = new HeroUnit();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java [hunk 3]
```diff
@@ -4,10 +4,42 @@
...
+import lombok.Getter;
 
 public abstract class PageWithPosts extends WikiBasePageObject {
 
+  public abstract SignInToFollowModalDialog getSignInDialog();
+
+  public abstract PageWithPosts open();
+
+  @Getter(lazy = true)
+  private final SortingFilterDesktop sortingFilterDesktop = new SortingFilterDesktop();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java [hunk 4]
```diff
@@ -4,10 +4,42 @@
...
+  private final Promoting promoting = new Promoting();
+
+  @Getter(lazy = true)
+  private final CommunityBadge communityBadge = new CommunityBadge();
+
+  @Getter(lazy = true)
+  private final HeroUnit heroUnit = new HeroUnit();
+
+  @Getter(lazy = true)
+  private final Moderation moderation = new Moderation();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java [hunk 5]
```diff
@@ -4,10 +4,42 @@
...
 
+  public abstract SignInToFollowModalDialog getSignInDialog();
+
+  public abstract PageWithPosts open();
+
+  @Getter(lazy = true)
+  private final SortingFilterDesktop sortingFilterDesktop = new SortingFilterDesktop();
+
+  @Getter(lazy = true)
+  private final DiscussionsHeader discussionsHeader = new DiscussionsHeader();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java [hunk 6]
```diff
@@ -4,10 +4,42 @@
...
+  private final CommunityBadge communityBadge = new CommunityBadge();
+
+  @Getter(lazy = true)
+  private final HeroUnit heroUnit = new HeroUnit();
+
+  @Getter(lazy = true)
+  private final Moderation moderation = new Moderation();
+
+  @Getter(lazy = true)
+  private final Post post = new Post();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java [hunk 7]
```diff
@@ -4,10 +4,42 @@
...
+  public abstract PageWithPosts open();
+
+  @Getter(lazy = true)
+  private final SortingFilterDesktop sortingFilterDesktop = new SortingFilterDesktop();
+
+  @Getter(lazy = true)
+  private final DiscussionsHeader discussionsHeader = new DiscussionsHeader();
+
+  @Getter(lazy = true)
+  private final FiltersPopOver filtersPopOver = new FiltersPopOver();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java [hunk 8]
```diff
@@ -4,10 +4,42 @@
...
+
+  @Getter(lazy = true)
+  private final Promoting promoting = new Promoting();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/SortingFilterDesktop.java [hunk 9]
```diff
@@ -0,0 +1,40 @@
+package com.wikia.webdriver.elements.mercury.components.discussions.desktop;
+
+import com.wikia.webdriver.common.logging.PageObjectLogging;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/SortingFilterDesktop.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.logging.PageObjectLogging'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/SortingFilterDesktop.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+import com.wikia.webdriver.elements.mercury.components.discussions.common.SortOption;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.BasePageObject;
+
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.Objects;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/desktop/SortingFilterDesktop.java]: MINOR Remove this unused import 'java.util.Objects'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 19 issues
* ![MAJOR][MAJOR] 5 major
* ![MINOR][MINOR] 14 minor

Watch the comments in this conversation to review them.

#### 9 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BasePageObject.java#L147](https://github.com/Wikia/selenium-tests/blob/ccb98b4ef6bb2f140894c943bc00aded092b03e7/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L147): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L163](https://github.com/Wikia/selenium-tests/blob/ccb98b4ef6bb2f140894c943bc00aded092b03e7/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L163): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L584](https://github.com/Wikia/selenium-tests/blob/ccb98b4ef6bb2f140894c943bc00aded092b03e7/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L584): Remove this unused private "getTabsCount" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [BasePageObject.java#L638](https://github.com/Wikia/selenium-tests/blob/ccb98b4ef6bb2f140894c943bc00aded092b03e7/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L638): Remove this unused "initialTabsNumber" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)
1. ![MAJOR][MAJOR] [SignupTests.java#L206](https://github.com/Wikia/selenium-tests/blob/ccb98b4ef6bb2f140894c943bc00aded092b03e7/src/test/java/com/wikia/webdriver/testcases/auth/SignupTests.java#L206): Remove this unused private "getEmailAlias" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [FollowPage.java#L4](https://github.com/Wikia/selenium-tests/blob/ccb98b4ef6bb2f140894c943bc00aded092b03e7/src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/FollowPage.java#L4): Remove this unused import 'com.wikia.webdriver.elements.mercury.components.discussions.common.Post'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [ReportedPostsAndRepliesPage.java#L4](https://github.com/Wikia/selenium-tests/blob/ccb98b4ef6bb2f140894c943bc00aded092b03e7/src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/ReportedPostsAndRepliesPage.java#L4): Remove this unused import 'com.wikia.webdriver.elements.mercury.components.discussions.common.Post'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [UserPostsPage.java#L6](https://github.com/Wikia/selenium-tests/blob/ccb98b4ef6bb2f140894c943bc00aded092b03e7/src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/UserPostsPage.java#L6): Remove this unused import 'com.wikia.webdriver.elements.mercury.components.discussions.common.Post'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [WikiBasePageObject.java#L41](https://github.com/Wikia/selenium-tests/blob/ccb98b4ef6bb2f140894c943bc00aded092b03e7/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java#L41): Remove this unused import 'java.time.Duration'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 9 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BasePageObject.java#L147](https://github.com/Wikia/selenium-tests/blob/59a086cdf1ea2baedfaf94f3a75064f653d54af7/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L147): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L163](https://github.com/Wikia/selenium-tests/blob/59a086cdf1ea2baedfaf94f3a75064f653d54af7/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L163): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L584](https://github.com/Wikia/selenium-tests/blob/59a086cdf1ea2baedfaf94f3a75064f653d54af7/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L584): Remove this unused private "getTabsCount" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [BasePageObject.java#L638](https://github.com/Wikia/selenium-tests/blob/59a086cdf1ea2baedfaf94f3a75064f653d54af7/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L638): Remove this unused "initialTabsNumber" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)
1. ![MAJOR][MAJOR] [SignupTests.java#L206](https://github.com/Wikia/selenium-tests/blob/59a086cdf1ea2baedfaf94f3a75064f653d54af7/src/test/java/com/wikia/webdriver/testcases/auth/SignupTests.java#L206): Remove this unused private "getEmailAlias" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [FollowPage.java#L4](https://github.com/Wikia/selenium-tests/blob/59a086cdf1ea2baedfaf94f3a75064f653d54af7/src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/FollowPage.java#L4): Remove this unused import 'com.wikia.webdriver.elements.mercury.components.discussions.common.Post'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [ReportedPostsAndRepliesPage.java#L4](https://github.com/Wikia/selenium-tests/blob/59a086cdf1ea2baedfaf94f3a75064f653d54af7/src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/ReportedPostsAndRepliesPage.java#L4): Remove this unused import 'com.wikia.webdriver.elements.mercury.components.discussions.common.Post'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [UserPostsPage.java#L6](https://github.com/Wikia/selenium-tests/blob/59a086cdf1ea2baedfaf94f3a75064f653d54af7/src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/UserPostsPage.java#L6): Remove this unused import 'com.wikia.webdriver.elements.mercury.components.discussions.common.Post'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [WikiBasePageObject.java#L41](https://github.com/Wikia/selenium-tests/blob/59a086cdf1ea2baedfaf94f3a75064f653d54af7/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java#L41): Remove this unused import 'java.time.Duration'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/testcases/discussions/SortingTests.java [hunk 11]
```diff
@@ -23,54 +23,74 @@
...
-    filters.clickTrendingOption();
-    new Loading(driver).handleAsyncPageReload();
+  private void performSortingCheckOnMobile(PageWithPosts page, SortOption sortOption) {
+    page
+      .getDiscussionsHeader()
+      .openFilterMenu()
+      .chooseSortingOption(sortOption)
+      .clickApplyButton()
+      .waitForPageReload();
+    Assertion.assertTrue(page.getCurrentUrl().contains(sortOption.getQuery()));
```
Comment (sqreek) [inline src/test/java/com/wikia/webdriver/testcases/discussions/SortingTests.java]: is this sufficient check?

File: src/test/java/com/wikia/webdriver/testcases/discussions/SortingTests.java [hunk 12]
```diff
@@ -23,54 +23,74 @@
...
+  }
 
-    Assertion.assertTrue(driver.getCurrentUrl().contains("trending"));
+  private void performSortingCheckOnDesktop(PageWithPosts page, SortOption sortOption) {
+    page
+      .getSortingFilterDesktop()
+      .chooseSortingOption(sortOption)
+      .waitForPageReload();
+    Assertion.assertTrue(page.getCurrentUrl().contains(sortOption.getQuery()));
```
Comment (sqreek) [inline src/test/java/com/wikia/webdriver/testcases/discussions/SortingTests.java]: is this sufficient check?

File: src/test/java/com/wikia/webdriver/testcases/discussions/SortingTests.java (hunk 11 above)
Comment (tortila) [reply src/test/java/com/wikia/webdriver/testcases/discussions/SortingTests.java]: As discussed offline, the only way we could safely perform checking if posts are sorted is to have a dedicated wiki with known state of posts. I can create a separate task to do it, but to tell the truth, I don't see much value. I'd rather make sure this is covered by tests on backend side.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 9 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BasePageObject.java#L147](https://github.com/Wikia/selenium-tests/blob/dcf1bc559a2f6e2843208c3fe271f68830814263/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L147): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L163](https://github.com/Wikia/selenium-tests/blob/dcf1bc559a2f6e2843208c3fe271f68830814263/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L163): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L584](https://github.com/Wikia/selenium-tests/blob/dcf1bc559a2f6e2843208c3fe271f68830814263/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L584): Remove this unused private "getTabsCount" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [BasePageObject.java#L638](https://github.com/Wikia/selenium-tests/blob/dcf1bc559a2f6e2843208c3fe271f68830814263/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L638): Remove this unused "initialTabsNumber" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)
1. ![MAJOR][MAJOR] [SignupTests.java#L206](https://github.com/Wikia/selenium-tests/blob/dcf1bc559a2f6e2843208c3fe271f68830814263/src/test/java/com/wikia/webdriver/testcases/auth/SignupTests.java#L206): Remove this unused private "getEmailAlias" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [FollowPage.java#L4](https://github.com/Wikia/selenium-tests/blob/dcf1bc559a2f6e2843208c3fe271f68830814263/src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/FollowPage.java#L4): Remove this unused import 'com.wikia.webdriver.elements.mercury.components.discussions.common.Post'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [ReportedPostsAndRepliesPage.java#L4](https://github.com/Wikia/selenium-tests/blob/dcf1bc559a2f6e2843208c3fe271f68830814263/src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/ReportedPostsAndRepliesPage.java#L4): Remove this unused import 'com.wikia.webdriver.elements.mercury.components.discussions.common.Post'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [UserPostsPage.java#L6](https://github.com/Wikia/selenium-tests/blob/dcf1bc559a2f6e2843208c3fe271f68830814263/src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/UserPostsPage.java#L6): Remove this unused import 'com.wikia.webdriver.elements.mercury.components.discussions.common.Post'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [WikiBasePageObject.java#L41](https://github.com/Wikia/selenium-tests/blob/dcf1bc559a2f6e2843208c3fe271f68830814263/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java#L41): Remove this unused import 'java.time.Duration'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**